test/testasyncio.c

```diff
@@ -0,0 +1,176 @@
+/*
+  Copyright (C) 1997-2024 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely.
+*/
+
+#define SDL_MAIN_USE_CALLBACKS 1
+#include <SDL3/SDL_main.h>
+#include <SDL3/SDL_test.h>
+#include <SDL3/SDL_test_common.h>
+
+static SDL_Renderer *renderer = NULL;
+static SDL_Texture *texture = NULL;
+static SDL_IOAsyncGroup *group = NULL;
+static SDLTest_CommonState *state = NULL;
+
+SDL_AppResult SDL_AppInit(void **appstate, int argc, char *argv[])
```

Comment:
Does it make sense to add some testautomation tests to validate parameter checking and invalid operations?

Reply:
Probably, but testautomation intimidates the crap out of me.  :)